ruby/red-arrow/lib/arrow/table.rb

```diff
@@ -197,10 +197,38 @@ def each_record_batch
     alias_method :length, :n_rows
 
     alias_method :[], :find_column
+    alias_method :slice_raw, :slice
 
-    # TODO
+    # @overload slice(offset, length)
     #
-    # @return [Arrow::Table]
+    #   @param offset [Integer] The offset of sub Arrow::Table.
+    #   @param length [Integer] The length of sub Arrow::Table.
+    #   @return [Arrow::Table]
+    #     The sub Arrow::Table. It covers only from
+    #     `offset` to `offset + length` range.
+    #
+    # @overload slice(offset)
+    #
+    #   @param offset [Integer] The offset of sub Arrow::Table.
+    #   @return [Arrow::Table]
+    #      The sub Arrow::Table. It covers only from
+    #      `offset` until end of the table.
```

Comment:
@kou This breaks backward compatibility. But I've changed it because I think it better to match the C++ API.
How do you think about this?  Is there anything we need to do to Red Arrow users to make this change?

Reply:
This is expected to be the same interface as Ruby's `Array#slice(index)`: https://docs.ruby-lang.org/en/2.6.0/Array.html#method-i-slice

But the current implementation doesn't match the interface... We should return `Arrow::Record` or `nil` for `slice(index)`.

The current API accepts argument and block mix: `table.slice(1) {...}`. But we should stop this feature to implement `slice(index) -> Record|nil`. It breaks backward compatibility but is acceptable.

```ruby
if block_given?
  unless args.empty?
    raise ArgumentError, ...
  end
  block_slicer = yield(...)
  ...
else
  case args.size
  ...
  end
end
```

Reply:
How about endless range `1..` to return a table that covers from `offset` until the end of the table?

Reply:
I see. I've returned `Arrow::Record` or `nil` for `slice(index)`.